### ISSUE TITLE
Backport bugfixes from trunk to 2.1 branch

### ISSUE THREAD

**Issue by gcode-importer** [uclouvain/openjpeg#405]
Originally reported on Google Code with ID 405
```
compile bugfixes since last release and backport to maintained branches
```

Reported by [detonin](https://github.com/detonin) on 2014-09-30 12:22:05

**Comment by mayeut** [comment]
@detonin,

Isn't master the same as branch 2.1 ( need to check ABI/API #306 ) ?

**Comment by detonin** [comment]
@mayeut I merged master in 2.1 branch. According to https://github.com/uclouvain/openjpeg/wiki/DevelopmentProcess, we should actually have a 1.x.y, and a 2.x.y branch, instead of having different branches for minor versions bumps. Anyway, master always corresponds to the latest dev and merging into 2.1 makes sense as we are about to break or at least enhance the API in the next release.
